### PR TITLE
[Merged by Bors] - refactor for passing logger and data dir to p2p from main

### PR DESCRIPTION
**Pull request by y0sher**
## Motivation
we had long time tech debt in p2p, logs and directory path were separated and used global variables.

## Changes
This is mainly a refactor, some other stuff -
- Bringing back tests (this is still in the works)
- Passing the logger from main to P2P and add functionality we have for other loggers (change level..)
- passing the Datadir to p2p and initializing any files in there under the `p2p` folder.
- added context in SelectPeers to enable it to stop when shutting down.

## Test Plan
re-enabled p2p integration tests.

### CODE REVIEW

File: filesystem/dirs_test_helpers.go [hunk 1]
```diff
@@ -3,63 +3,57 @@ package filesystem
...
-	// remove
-	err = os.RemoveAll(aPath)
-	if err != nil {
-		t.Fatalf("Failed to delete spacemesh data dir: %s", err)
-	}
-
-	config.ConfigValues.DataFilePath = "~/.spacemesh"
-}
+//
+//// SetupTestSpacemeshDataFolders sets up a data folder to this specific test
```
Comment (y0sher) [inline filesystem/dirs_test_helpers.go]: should delete this before merging

File: cmd/node/node.go [hunk 2]
```diff
@@ -579,7 +577,7 @@ func (app *SpacemeshApp) checkTimeDrifts() {
 			_, err := timesync.CheckSystemClockDrift()
 			if err != nil {
 				app.log.Error("System time couldn't synchronize %s", err)
-				app.stopServices()
+				cmdp.Cancel()
```
Comment (y0sher) [inline cmd/node/node.go]: I think this is already done like this in develop, so will be gone on rebase

File: cmd/node/node.go [hunk 3]
```diff
@@ -770,7 +768,15 @@ func (app *SpacemeshApp) getIdentityFile() (string, error) {
 }
 
 func (app *SpacemeshApp) Start(cmd *cobra.Command, args []string) {
-	log.Event().Info("Starting Spacemesh")
+	log.With().Info("Initializing Spacemesh", log.String("data-dir", app.Config.DataDir), log.String("post-dir", app.Config.POST.DataDir))
```
Comment (antonlerner) [inline cmd/node/node.go]: why change this print? we rely on this text in our searches, you can add more parameters to it but I wouldn't change the main title

File: p2p/discovery/discovery.go [hunk 4]
```diff
@@ -128,28 +132,31 @@ func (d *Discovery) Update(addr, src *node.NodeInfo) {
 
 // TODO: Replace `node.LocalNode` with `NodeInfo` and `log.Log`.
```
Comment (antonlerner) [inline p2p/discovery/discovery.go]: remove this todo?

File: p2p/discovery/protocol.go [hunk 5]
```diff
@@ -46,17 +49,17 @@ const PINGPONG = 0
...
-		local:     local,
+		local:     &node.NodeInfo{ID: local.Array(), IP: net.IPv4zero, ProtocolPort: 7513, DiscoveryPort: 7513},
 		table:     rt,
 		msgServer: s,
 		logger:    log,
 	}
 
-	//tcp := &net.TCPAddr{local.IP, int(local.ProtocolPort), ""}
-	//udp := &net.UDPAddr{local.IP, int(local.DiscoveryPort), ""}
+	//tcp := &net.TCPAddr{localNode.IP, int(localNode.ProtocolPort), ""}
```
Comment (antonlerner) [inline p2p/discovery/protocol.go]: do you need this code?

File: p2p/gossip/protocol.go [hunk 6]
```diff
@@ -202,6 +202,16 @@ func (prot *Protocol) getPriority(protoName string) priorityq.Priority {
 	return v
 }
 
+func (prot *Protocol) drainPropagationQueue() {
```
Comment (antonlerner) [inline p2p/gossip/protocol.go]: is this really needed? please revise...

File: p2p/swarm.go [hunk 7]
```diff
@@ -104,34 +106,59 @@ func (s *swarm) waitForGossip() error {
 
 // newSwarm creates a new P2P instance, configured by config, if newNode is true it will create a new node identity
 // and not load from disk. it creates a new `net`, connection pool and discovery.
-func newSwarm(ctx context.Context, config config.Config, newNode bool, persist bool) (*swarm, error) {
+func newSwarm(ctx context.Context, config config.Config, logger log.Log, datadir string) (*swarm, error) {
+
+	if logger == log.NilLogger {
```
Comment (antonlerner) [inline p2p/swarm.go]: why have this check? why not decide what kind of logger this func uses by sending it to the func

File: p2p/swarm.go [hunk 8]
```diff
@@ -104,34 +106,59 @@ func (s *swarm) waitForGossip() error {
...
-		l, err = node.NewLocalNode(config, &addr, persist)
+	// Load an existing identity from file if exists.
+	if config.NodeID != "" {
+		l, err = node.LoadIdentity(datadir, config.NodeID)
+		if err != nil {
+			return nil, err
+		}
 	}
 
+	l, err = node.NewNodeIdentity()
```
Comment (antonlerner) [inline p2p/swarm.go]: this will override `l` always regardless of the if in line 123

**Review by antonlerner — APPROVED** [review]

File: p2p/gossip/protocol.go (hunk 6 above)
Comment (y0sher) [reply p2p/gossip/protocol.go]: probably not, was a desperate attempt to fix travis.. going to remove now.

**Comment by y0sher** [comment]
bors merge

**Comment by y0sher** [comment]
bors merge

**Comment by y0sher** [comment]
bors merge

**Comment by y0sher** [comment]
bors merge

**Comment by y0sher** [comment]
bors merge

**Comment by y0sher** [comment]
bors merge

**Comment by y0sher** [comment]
bors merge p=100

**Comment by y0sher** [comment]
bors merge p=10

**Comment by y0sher** [comment]
bors merge p=1

**Comment by y0sher** [comment]
bors merge

**Pull request closed without merging**